### PR TITLE
feat(provider) INT-5031: Stripe UPE Payment Method

### PR DESCRIPTION
**Pull request by BC-PabloRivera**
What? [INT-5031](https://jira.bigcommerce.com/browse/INT-5031)
Stripe Payment Element with credit card support

Why?
To create the payment method for stripeUPE

## Testing / Proof
image
image

## Depends on
https://github.com/bigcommerce/checkout-sdk-js/pull/1295

@bigcommerce/checkout

### CODE REVIEW

**Review by JoseAlvarezHdz — APPROVED** [review]
LGTM 👍🏾

File: src/app/payment/paymentMethod/StripeUPEPaymentMethod.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    return <>
+        <HostedWidgetPaymentMethod
+            { ...rest }
+            containerId={ containerId }
+            hideContentWhenSignedOut
+            initializePayment={ initializeStripePayment }
+            method={ method }
+        />
+        {
+            method.id === 'iban' &&
```
Comment (icatalina) [inline src/app/payment/paymentMethod/StripeUPEPaymentMethod.tsx:44]: This is a bit weird, we say in line 27 and 16 that `method.id` will always be card, then here, we compare with `iban`.

File: src/app/payment/paymentMethod/StripeUPEPaymentMethod.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    method,
+    storeUrl,
+    ...rest
+    }) => {
+    const paymentMethodType = method.id as StripeElementType;
+    const containerId = `stripe-${paymentMethodType}-component-field`;
+
+    const initializeStripePayment: HostedWidgetPaymentMethodProps['initializePayment'] = useCallback(async (options: PaymentInitializeOptions) => {
+        return initializePayment({ ...options,
+            stripeupe: { containerId}});
```
Comment (icatalina) [inline src/app/payment/paymentMethod/StripeUPEPaymentMethod.tsx:32]: Can we get the formatting here a bit more aligned with the rest of the repo?
```suggestion
        return initializePayment({
            ...options,
            stripeupe: { containerId },
        });
```

**Comment by BC-PabloRivera** [comment]
♻️ I addressed your comments @icatalina
please review again

File: src/app/payment/paymentMethod/StripeUPEPaymentMethod.tsx [hunk 3]
```diff
@@ -40,12 +41,6 @@ const StripeUPEPaymentMethod: FunctionComponent<StripePaymentMethodProps & WithI
             initializePayment={ initializeStripePayment }
             method={ method }
         />
-        {
-            method.id === 'iban' &&
-            <p className="stripe-sepa-mandate-disclaimer">
-                <TranslatedString data={ {storeUrl} } id="payment.stripe_sepa_mandate_disclaimer" />
-            </p>
-        }
```
Comment (icatalina) [inline src/app/payment/paymentMethod/StripeUPEPaymentMethod.tsx]: This wasn't necessary? why was it there in the original pr?

**Review by icatalina — APPROVED** [review]

**Review by TomA-R — APPROVED** [review]

**Pull request merged**